woocommerce/woocommerce-ios
# [Woo POS] [Cash & Receipts] Disallow sending receipt and show toast when WC version is not eligible

Closes: #14465


## Description
This PR adds eligibility checks to receipts usage, prompting the merchant to upgrade to WooCommerce 9.5 to generate receipts in POS.

We add a toast/notification to the "success payment" view displaying that message, which they can tap to dismiss.

https://github.com/user-attachments/assets/5d467730-aa1a-4829-85f4-663439cb79ed


## Testing information
- Set feature flag `.sendReceiptsForPointOfSale` to true
- Log into Povilas testing site `https://lovely-aware-sturgeon.jurassic.ninja/` details: PdfdoF-5Qj-p2
- Proceed with a POS transaction
- Observe that the modal to enter email for receipts works normally
- Switch to a non eligible site (Woo <9.5) or edit `minimumVersionForPOSReceipts = 10`
- Proceed with a POS transaction
- Observe that upon tapping `Receipt` the toast shows up 
- Tapping the toast should dismiss it.

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
